dotnet/src/SemanticKernel.Skills/Skills.Memory.CosmosDB/CosmosMemoryStore.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Memory.Collections;
+using Microsoft.SemanticKernel.Memory.Storage;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Skills.Memory.CosmosDB;
+public class CosmosMemoryStore<TEmbedding> : IMemoryStore<TEmbedding>, IDisposable
+    where TEmbedding : unmanaged
+{
+    private bool _disposedValue;
+
+    private CosmosClient _client;
+    private string _databaseName;
+    private string _containerName;
+
+    public CosmosMemoryStore(CosmosClient client, string databaseName, string containerName)
+    {
+        this._client = client;
+        this._databaseName = databaseName;
+        this._containerName = containerName;
+    }
+
+    public async Task<DataEntry<IEmbeddingWithMetadata<TEmbedding>>?> GetAsync(string collection, string key, CancellationToken cancel = default)
```

Comment:
Please add <inheritdoc/> tags to all public methods inheriting from the interface.